server/src/main/java/org/opensearch/indices/replication/SegmentReplicator.java

```diff
@@ -102,6 +110,92 @@ SegmentReplicationTarget startReplication(
         return target;
     }
 
+    public ReplicationStats getSegmentReplicationStats(final ShardId shardId) {
```

Comment:
can you pls add javadoc for these methods with a brief description of the computation.